cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/BasinEndpointInput.java

```diff
@@ -0,0 +1,148 @@
+        private final String projectId;
+        private final String officeId;
+
+        private GetAll(String projectId, String officeId) {
+            this.projectId = Objects.requireNonNull(projectId, "Project Id required for getAll basin endpoint");
+            this.officeId = Objects.requireNonNull(officeId, "Project office Id required for getAll basin endpoint");
+        }
+
+        @Override
+        public HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryParameter(PROJECT_ID_QUERY_PARAMETER, projectId)
+                    .addQueryParameter(OFFICE_QUERY_PARAMETER, officeId)
+                    .addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1);
+        }
+    }
+
+    public static final class GetOne extends EndpointInput {
+        public static final String OFFICE_QUERY_PARAMETER = "office";
+        private final String officeId;
+        private final String basinId;
+
+        public GetOne(String basinId, String officeId) {
+            this.basinId = Objects.requireNonNull(basinId, "Basin Id required for getOne basin endpoint");
+            this.officeId = Objects.requireNonNull(officeId, "Basin office Id required for getOne basin endpoint");
+        }
+
+        public String basinId() {
+            return basinId;
+        }
+
+        @Override
+        public HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryParameter(OFFICE_QUERY_PARAMETER, officeId)
+                    .addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1);
+        }
+    }
+
+    public static final class Post extends EndpointInput {
+        private final Basin basin;
+
+        public Post(Basin basin) {
+            this.basin = Objects.requireNonNull(basin,
+                    "Cannot access the basin POST endpoint without a basin value");
+        }
+
+        public Basin basin() {
+            return basin;
+        }
+
+        @Override
+        public HttpRequestBuilder addInputParameters(HttpRequestBuilder httpRequestBuilder) {
+            return httpRequestBuilder.addQueryHeader(ACCEPT_QUERY_HEADER, ACCEPT_HEADER_V1);
+        }
+    }
+
+    public static final class Delete extends EndpointInput {
+        static final String OFFICE_QUERY_PARAMETER = "office";
+        static final String METHOD_QUERY_PARAMETER = "method";
+        private final String basinId;
+        private final String officeId;
+        private DeleteMethod deleteMethod;
+
+        public Delete(String basinId, String officeId) {
+            this.basinId = Objects.requireNonNull(basinId, "Cannot access the basin DELETE endpoint without a basin id");
+            this.officeId = Objects.requireNonNull(officeId, "Cannot access the embankment DELETE endpoint without an office id");
+        }
+
+        public void setDeleteMethod(DeleteMethod deleteMethod) {
```

Comment:
Is a DeleteMethod required? If so, should move this into the Delete's constructor and make it final. 

If it's not required, use a builder method here, instead of a setter.

Reply:
Addressed in latest push

Reply:
Looks like you moved this into the constructor, which tells me its required. Add a Objects.requireNonNull with a message like the other required fields.

Reply:
Fixed in latest push